x/gamm/types/pool.go

```diff
@@ -126,6 +122,19 @@ type PoolAmountOutExtension interface {
 	IncreaseLiquidity(sharesOut sdk.Int, coinsIn sdk.Coins)
 }
 
+// WeightedPoolExtension is an extension of the PoolI interface
+// That defines an additional API for handling the pool's weights.
+type WeightedPoolExtension interface {
+	PoolI
+
+	// PokePool determines if a pool's weights need to be updated and updates
+	// them if so.
+	PokePool(blockTime time.Time)
```

Comment:
Why is poke pool on this? Its more general than for a weighted pool?

Reply:
According to the [spec](https://github.com/osmosis-labs/osmosis/blob/babeebe957986ed13b6a3e7b2dc01f04da876f10/x/gamm/pool-models/balancer/pool.go#L395-L396) and balancer implementation, `PokePool` is only concerned with weights.

Stableswap did not have this method implemented, and I don't see why we would need it since there are no weights. So it seems to be weight-specific

Reply:
It seems like our current implementation of PokePool only concerns itself with updating weights. Is there some extension of PokePool that is planned that would make it useful for stableswap/concentrated-liquidity pools?